clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareFetchMetricsAggregator.java

```diff
@@ -27,7 +28,7 @@ public class ShareFetchMetricsAggregator {
 
     public ShareFetchMetricsAggregator(ShareFetchMetricsManager shareFetchMetricsManager, Set<TopicPartition> partitions) {
         this.shareFetchMetricsManager = shareFetchMetricsManager;
-        this.unrecordedPartitions = partitions;
+        this.unrecordedPartitions = new HashSet<>(partitions);
```

Comment:
I'm not sure creating a set from a set parameter is ideal. The test failures are because the tests are creating a Collections.singleton and passing that as the set. Because that's not mutable, the tests fail. So, I suggest changing the tests is part of the fix.

Reply:
Okay yeah makes sense, I have changed the way the constructor was called in the tests now.